frontend/src/container/TopNav/AutoRefreshV2/index.tsx

```diff
@@ -0,0 +1,201 @@
+
+import { getMinMax, options } from './config';
+import { ButtonContainer } from './styles';
+
+function AutoRefresh({
+	disabled = false,
+	showAutoRefreshBtnPrimary = true,
+}: AutoRefreshProps): JSX.Element {
+	const globalTime = useSelector<AppState, GlobalReducer>(
+		(state) => state.globalTime,
+	);
+	const { pathname } = useLocation();
+
+	const isDisabled = useMemo(
+		() =>
+			disabled ||
+			globalTime.isAutoRefreshDisabled ||
+			globalTime.selectedTime === 'custom',
+		[globalTime.isAutoRefreshDisabled, disabled, globalTime.selectedTime],
+	);
+
+	const localStorageData = JSON.parse(get(DASHBOARD_TIME_IN_DURATION) || '{}');
+
+	const localStorageValue = useMemo(() => localStorageData[pathname], [
+		pathname,
+		localStorageData,
+	]);
+
+	const [isAutoRefreshEnabled, setIsAutoRefreshfreshEnabled] = useState<boolean>(
+		Boolean(localStorageValue),
+	);
+
+	const dispatch = useDispatch<Dispatch<AppActions>>();
+
+	useEffect(() => {
+		const isAutoRefreshEnabled = Boolean(localStorageValue);
+		dispatch({
+			type: UPDATE_AUTO_REFRESH_INTERVAL,
+			payload: localStorageValue,
+		});
+		setIsAutoRefreshfreshEnabled(isAutoRefreshEnabled);
+	}, [localStorageValue, dispatch]);
+
+	const params = useUrlQuery();
+
+	const [selectedOption, setSelectedOption] = useState<string>(
+		localStorageValue || options[0].key,
+	);
+
+	useEffect(() => {
+		setSelectedOption(localStorageValue || options[0].key);
+	}, [localStorageValue, params]);
+
+	const getOption = useMemo(
+		() => options.find((option) => option.key === selectedOption),
+		[selectedOption],
+	);
+
+	useInterval(() => {
+		const selectedValue = getOption?.value;
+
+		if (isDisabled || !isAutoRefreshEnabled) {
+			return;
+		}
+
+		if (selectedOption !== 'off' && selectedValue) {
+			const { maxTime, minTime } = getMinMax(
+				globalTime.selectedTime,
+				globalTime.minTime,
+				globalTime.maxTime,
+			);
+
+			dispatch({
+				type: UPDATE_TIME_INTERVAL,
+				payload: {
+					maxTime,
+					minTime,
+					selectedTime: globalTime.selectedTime,
+				},
+			});
+		}
+	}, getOption?.value || 0);
+
+	const onChangeHandler = useCallback(
+		(selectedValue: string) => {
+			setSelectedOption(selectedValue);
+			params.set(DASHBOARD_TIME_IN_DURATION, selectedValue);
+			set(
+				DASHBOARD_TIME_IN_DURATION,
+				JSON.stringify({ ...localStorageData, [pathname]: selectedValue }),
+			);
+			setIsAutoRefreshfreshEnabled(true);
+		},
+		[params, pathname, localStorageData],
+	);
+
+	const onChangeAutoRefreshHandler = useCallback(
+		(event: CheckboxChangeEvent) => {
+			const { checked } = event.target;
+			if (!checked) {
+				// remove the path from localstorage
+				set(
+					DASHBOARD_TIME_IN_DURATION,
+					JSON.stringify(_omit(localStorageData, pathname)),
+				);
+			}
+			setIsAutoRefreshfreshEnabled(checked);
+		},
+		[localStorageData, pathname],
+	);
+
+	if (globalTime.selectedTime === 'custom') {
+		// eslint-disable-next-line react/jsx-no-useless-fragment
+		return <></>;
```

Comment:
Check if we can avoid this.

Reply:
this is part of existing code , will refactor a bit in next PR will check if we can remove this